### ISSUE TITLE
Don't use vmap(loss) ever, especially not with dropout or other rng

### ISSUE THREAD

**Issue by Ivan-Zhou** [stanford-crfm/levanter#226]
vmap(dropout) when using `unsafe_rbg` (i.e. "fast random number generation") is very slow, compared to using dropout with a batched shape: up to 10x slower. This is unfortunately not possible to fix (per Roy)

We currently generate 1 random key per example in train_lm and vmap over the example(s) and the key(s). Doing it this way has the nice perk that each example gets a distinct, reproducible random state even in the presence of grad accum... But it's kind of a red herring b/c floating point math means it's not the same results anyway.

So we need to stop doing that.

Instead we need to not use vmap over the batch and have grad accum split the key. We can probably hack in the specialness of `key` since it's so special.

# Old Issue

I observed massive negative impact on the training throughput by `unsafe_rbg`:
- With `unsafe_rbg`: 68.5K tokens per sec
- Without `unsafe_rbg`: 537.7K tokens per sec
![image](https://github.com/stanford-crfm/levanter/assets/8097756/39f4b836-5859-4fe3-8199-c93e1303a236)

The way I enabled `unsafe_rbg` is by adding the following line to the beginning of `src/levanter/main/train_lm.py` ([commit](https://github.com/stanford-crfm/levanter/commit/6ad785de086e491ff27da46076d92849b7d0df7d))
```
jax.config.update("jax_default_prng_impl", "unsafe_rbg")
```

**Comment by dlwh** [comment]
Minimized. will keep this open until they fix: https://github.com/google/jax/issues/16792

**Comment by dlwh** [comment]
They're not able to fix, so we need to stop using vmap(loss) in train_lm. changing the issue to reflect that
